
syntax = "proto3";


package helloworld;

// service definition
service Greeter {
    rpc SayHello (HelloRequest) returns (HelloReply) {}

    // Sends multiple greetings
    rpc ManyHello (ManyHelloRequest) returns (stream HelloReply) {}

}

// message definition
message HelloRequest {
    string name = 1;
}

message HelloReply {
    string message = 1;
}

message ManyHelloRequest {
    string name = 1;
    int32 num_times = 2;
}